code/datums/components/barber.dm

```diff
@@ -63,8 +62,30 @@ TYPEINFO(/datum/component/barber)
 	initialization_args = list()
 
 /datum/component/barber
+	var/list/all_hairs = list()
```

Comment:
This should definitely not be re-initialized on every component, make it static or something.